metabase/metabase
# Extend `addUndo` API for auto-apply filters toast

Epic: https://github.com/metabase/metabase/issues/29267

### Description

This PR extends `addUndo` API, which is necessary for allowing toasts created via `addUndo` to not contain any timeout which is specced in [[Epic] Prevent auto-apply filters in slow dashboards](https://github.com/metabase/metabase/issues/29267#top)

### How to verify
Follow the test step in https://github.com/metabase/metabase/pull/30371, but both
1. Dashboard slow toast and
2. Auto-apply filters toast
should have no timeout, and only disappear when users interact with the toast. But please note that the dashboard slow toast can also be closed when all the dashboard cards finish loading.


### Checklist



- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/30525)
